✨(core) change field to store configuration name and improve fallback

rmoch opened the pull request:
Use 'lti_id' field instead of 'description' because it will ease course
based configuration fallback if this xblock is not installed

rmoch committed:
✨(core) change field to store configuration name and improve fallback

Use 'lti_id' field instead of 'description' because it will ease course
based configuration fallback if this xblock is not installed

sampaccoud commented on README.md:
```diff
@@ -1,10 +1,13 @@
+version.
+
+If this xblock is not installed or settings are not configured, courses using it
+will safely fall back on regular edX LTI consumer xblock.
```
fall back TO THE regular edX LTI consumer xblock?

jmaupetit replied:
`s/on/to the/`

sampaccoud commented on configurable_lti_consumer/configurable_lti_consumer.py:
```diff
@@ -28,32 +35,38 @@ def set_editable_field_names(self):
-        if "lti" in configuration:
+        configuration = self.get_configuration(self.lti_id)
+        if configuration and ("lti" in configuration):
             return configuration["lti"]["key"], configuration["lti"]["secret"]
```
Maybe we should stick to LTI's names for things:
- "lti" could be called something like "lti_passport_credentials",
- "key" should be called "oauth_consumer_key",
- "secret" should be called "shared_secret".

rmoch replied:
👍

jmaupetit commented on README.md:
```diff
@@ -1,10 +1,13 @@
+This XBlock inherits, enhances and replaces [edX LTI consumer Xblock](https://github.com/edx/xblock-lti-consumer) to make available to conceptors in CMS several preconfigured versions of
+LTI Consumer. Configurations are stored in Django settings instead of course
+advanced settings, and preconfigured fields are no longer modifiable for a given
+version.
```
I think a rebase went wrong and you re-write some parts I had already rewritten.

sampaccoud commented on configurable_lti_consumer/configurable_lti_consumer.py:
```diff
@@ -28,32 +35,38 @@ def set_editable_field_names(self):
+                        return conf[0]
+                    else:
+                        logger.warning(
+                            "Configuration '{name}' does not exist".format(name=name))
```
there should be an "else" clause to the "for" loop and raise an ConfigurationError as well because if the "for" loop completes normally, it means that we did not return and so the "lti_consumer" conf was not found.

sampaccoud commented on configurable_lti_consumer/configurable_lti_consumer.py:
```diff
@@ -28,32 +35,38 @@ def set_editable_field_names(self):
+                        logger.warning(
+                            "Configuration '{name}' does not exist".format(name=name))
+        else:
+            logger.warning("CONFIGURABLE_XBLOCKS_SETTINGS settings are not set")
```
let's break and raise a ConfigurationError instead of just logging a warning because it is the admin sys reponsibility to fix it before starting the server. WDYT?

sampaccoud commented on configurable_lti_consumer/configurable_lti_consumer.py:
```diff
@@ -28,32 +35,38 @@ def set_editable_field_names(self):
-                if conf:
-                    return conf[0]
-        return None
+        if getattr(settings, "CONFIGURABLE_XBLOCKS_SETTINGS"):
```
We could add a defaults.py file to this project to define the default setting (i.e act as a normal lti xblock):
```python
"""Default settings for the ``core`` app of the Marsha project."""
from django.conf import settings


CONFIGURABLE_XBLOCKS_SETTINGS = getattr(
    settings, "CONFIGURABLE_XBLOCKS_SETTINGS", {
        "components": [{
            "module": "lti_consumer",
            "base_class": "ConfigurableLtiConsumerXBlock",
            "subclasses": [{
                "name": "Generic",
                "display": "Generic LTI xblock",
                "default_values": {
                    "description": "Generic",
                    "ask_to_send_username": True,
                    "ask_to_send_email": True
                }
            }]
        }]
    }
)
```

This would allow us to remove the default settings in [openfun/openedx-docker](https://github.com/openfun/openedx-docker)

rmoch replied:
Good idea, then it implies that this configuration becomes specific to this xblock. 
the `components` list becomes useless

rmoch committed:
🐳(dev) add working environment

Developers working on this project need an easy way to setup a
development environment for this xblock. Thanks to docker and
docker-compose, we are now able to easily bootstrap such environment in
an openedx context.
